x/appchain/coordinator/keeper/impl_epochs_hooks.go

```diff
@@ -0,0 +1,79 @@
+package keeper
+
+import (
+	avstypes "github.com/ExocoreNetwork/exocore/x/avs/types"
+	epochstypes "github.com/ExocoreNetwork/exocore/x/epochs/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// EpochsHooksWrapper is the wrapper structure that implements the epochs hooks for the dogfood
+// keeper.
+type EpochsHooksWrapper struct {
+	keeper *Keeper
+}
+
+// Interface guard
+var _ epochstypes.EpochHooks = EpochsHooksWrapper{}
+
+// EpochsHooks returns the epochs hooks wrapper. It follows the "accept interfaces, return
+// concretes" pattern.
+func (k *Keeper) EpochsHooks() EpochsHooksWrapper {
+	return EpochsHooksWrapper{k}
+}
+
+// AfterEpochEnd is called after an epoch ends. It is called during the BeginBlock function.
+func (wrapper EpochsHooksWrapper) AfterEpochEnd(
+	ctx sdk.Context, identifier string, epoch int64,
+) {
+	// whenever an epoch ends, we should iterate over the list of pending subscriber chains
```

Comment:
So we don't wait for the end of Dogfood's epoch to activate the registered app chains?

Reply:
App chains are free to choose their own epoch identifier (day, week, hour, minute). It does not have to be the same as that of x/dogfood. This is the same as any AVS.